### PR TITLE
test: e2e Reveal Secret Recovery Phrase Quiz answer confirmation

### PR DESCRIPTION
**Pull request by plasmacorral**
## **Description**

This pull request introduces a new test that navigates through the Secret Recovery Phrase reveal quiz, up to the password entry prompt. It verifies that the flow can be successfully completed when incorrect answers are provided and then resubmitted with the correct answer chosen.

## **Related issues**

Fixes: [492](https://github.com/MetaMask/accounts-planning/issues/492)

## **Manual testing steps**

Not applicable

## **Screenshots/Recordings**

Not applicable

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by chrisleewilcox** [comment]
@plasmacorral you can look at swaps use of fixtures and see it that works for your tests.

./e2e/specs/swaps/swap-action-smoke.spec.js

**Review by cortisiko — CHANGES_REQUESTED** [review]
Good work on this 👏,  I have left some feedback

File: e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js [hunk 1]
```diff
@@ -34,6 +136,50 @@ class RevealSecretRecoveryPhrase {
   async enterPassword(password) {
     await Gestures.typeTextAndHideKeyboard(this.passwordInput, password);
   }
+
+  async tapGetStarted() {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js:140]: They were different, one was the `get` other was the `tap...`  Both will be removed from this file. TY!

File: e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js [hunk 2]
```diff
@@ -34,6 +136,50 @@ class RevealSecretRecoveryPhrase {
...
+
+  async tapContinueOne() {
+    await Gestures.waitAndTap(this.continueOne);
+  }
+
+  async tapQuestionTwoWrongAnswer() {
+    await Gestures.waitAndTap(this.questionTwoWrongAnswer);
+  }
+
+  async tapTryAgainTwo() {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js:164]: removing from this file. TY!

File: e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js [hunk 3]
```diff
@@ -34,6 +136,50 @@ class RevealSecretRecoveryPhrase {
...
+
+  async tapQuestionTwoRightAnswer() {
+    await Gestures.waitAndTap(this.questionTwoRightAnswer);
+  }
+
+  async tapContinueTwo() {
+    await Gestures.waitAndTap(this.continueTwo);
+  }
+
+  async tapTouchableBox() {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js:176]: Removed from this file. TY!

File: e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js [hunk 4]
```diff
@@ -34,6 +136,50 @@ class RevealSecretRecoveryPhrase {
...
+    await Gestures.waitAndTap(this.touchableBox);
+  }
+
+  async tapRevealRecoveryPhrase() {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js:180]: Removed from this file.  TY!

File: e2e/selectors/Settings/SecurityAndPrivacy/RevealSeedView.selectors.js [hunk 5]
```diff
@@ -13,4 +13,29 @@ export const RevealSeedViewSelectorsIDs = {
   SECRET_RECOVERY_PHRASE_LONG_PRESS_BUTTON_ID:
     'reveal-private-long-press-button',
   PASSWORD_INPUT_BOX_ID: 'private-credential-password-text-input',
+  SECRET_RECOVERY_PHRASE_QUIZ_GET_STARTED_BUTTON_ID: 'get-started-button',
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/selectors/Settings/SecurityAndPrivacy/RevealSeedView.selectors.js:16]: Done.  TY!

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        await loginToApp();
+        // Navigate settings to the Reveal Secret Recovery Phrase screen
+        await TabBarComponent.tapSettings();
+        await SettingsView.tapSecurityAndPrivacy();
+        await SecurityAndPrivacy.tapRevealSecretRecoveryPhraseButton();
+        //Start the quiz
+        await RevealSecretRecoveryPhrase.tapGetStarted();
+        //Answer the first question wrong
+        await RevealSecretRecoveryPhrase.tapQuestionOneWrongAnswer();
+        await TestHelpers.checkIfVisible(
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js:39]: Done.  TY!

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+          RevealSeedViewSelectorsIDs.SECRET_RECOVERY_PHRASE_QUIZ_QUESTION_ONE_WRONG_TITLE,
+        );
+        //Try again
+        await RevealSecretRecoveryPhrase.tapTryAgainOne();
+        //Answer the first question right
+        await RevealSecretRecoveryPhrase.tapQuestionOneRightAnswer();
+        await TestHelpers.checkIfVisible(
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js:46]: Done.  TY!

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+          RevealSeedViewSelectorsIDs.SECRET_RECOVERY_PHRASE_QUIZ_QUESTION_ONE_RIGHT_TITLE,
+        );
+        await RevealSecretRecoveryPhrase.tapContinueOne();
+        //Answer the second question wrong
+        await RevealSecretRecoveryPhrase.tapQuestionTwoWrongAnswer();
+        await TestHelpers.checkIfVisible(
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js:52]: Done.  TY!

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+          RevealSeedViewSelectorsIDs.SECRET_RECOVERY_PHRASE_QUIZ_QUESTION_TWO_WRONG_TITLE,
+        );
+        //Try again
+        await RevealSecretRecoveryPhrase.tapTryAgainTwo();
+        //Answer the second question right
+        await RevealSecretRecoveryPhrase.tapQuestionTwoRightAnswer();
+        await TestHelpers.checkIfVisible(
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js:59]: Done.  TY!

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  defaultGanacheOptions,
+} from '../../fixtures/fixture-helper';
+import FixtureBuilder from '../../fixtures/fixture-builder.js';
+
+describe(SmokeAccounts('Secret Recovery Phrase Quiz'), () => {
+  beforeAll(async () => {
+    await TestHelpers.reverseServerPort();
+  });
+
+  it('completes successfully after correcting wrong answers', async () => {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js]: Changed the wording a little, hopefully it is more clear?

File: e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js [hunk 11]
```diff
@@ -9,6 +9,108 @@ class RevealSecretRecoveryPhrase {
     );
   }
 
+  get getStarted() {
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/pages/Settings/SecurityAndPrivacy/RevealSecretRecoveryPhrase.js]: Restored this file to its previous state

File: e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        //Try again
+        await RevealSecretRecoveryPhrase.tapTryAgainTwo();
+        //Answer the second question right
+        await RevealSecretRecoveryPhrase.tapQuestionTwoRightAnswer();
+        await TestHelpers.checkIfVisible(
+          RevealSeedViewSelectorsIDs.SECRET_RECOVERY_PHRASE_QUIZ_QUESTION_TWO_RIGHT_TITLE,
+        );
+        await RevealSecretRecoveryPhrase.tapContinueTwo();
+        // land at reveal SRP view prompt for password
+        await TestHelpers.checkIfVisible(
```
[the remark this answers is not in the document]
Comment (plasmacorral) [reply e2e/specs/accounts/secret_recovery_phrase_quiz.spec.js]: Removed this for now.

**Review by cortisiko — APPROVED** [review]
LGTM ✅

**Review by gantunesr — APPROVED** [review]

**Pull request merged**